person.rb

```diff
@@ -0,0 +1,24 @@
+class Person
+  attr_reader :id, :parent_permission
+  attr_accessor :name, :age
+
+  def initialize(age, name = 'unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = !!parent_permission
+  end
+
+  def of_age?
+    is_of_age = false
+    is_of_age = true if @age >= 18
+    is_of_age
+  end
+  private :of_age?
+
+  def can_use_services?
+    can_use_services = false
+    can_use_services = true if @age >= 18 || @parent_permission == true
+    can_use_services
+  end
```

Comment:
- [ ] this is why the method `of_age?` is created, so you don't have to check if `@age` is larger or smaller than 18.
- [ ] [OPTIONAL] In this method it is not needed to first set a variable to `false` and then check with an if statement and return the variable, it can be written like below.
```
  def can_use_services?
    of_age? || @parent_permission == true
  end
```